hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/tools/MarkerTool.java

```diff
@@ -513,6 +526,11 @@ private ScanResult scan(
       final DirectoryPolicy filterPolicy)
       throws IOException, ExitUtil.ExitException {
 
+    // safety check: min and max are correctly ordered at this point.
+    Preconditions.checkArgument(minMarkerCount <= maxMarkerCount,
```

Comment:
nit: Is this check required again since we have already swapped above?

Reply:
no, but I'm just being extra paranoid

Reply:
it will harm nobody, we can leave this check here.